source/jetstream/iterator.go

```diff
@@ -0,0 +1,270 @@
+func (i *Iterator) Next(ctx context.Context) (sdk.Record, error) {
+	select {
+	case msg := <-i.messages:
+		sdkRecord, err := i.messageToRecord(msg)
+		if err != nil {
+			return sdk.Record{}, fmt.Errorf("convert message to record: %w", err)
+		}
+
+		if i.consumerInfo.Config.AckPolicy != nats.AckNonePolicy {
+			i.Lock()
+			i.unackMessages = append(i.unackMessages, msg)
+			i.Unlock()
+		}
+
+		return sdkRecord, nil
+
+	case <-ctx.Done():
+		return sdk.Record{}, ctx.Err()
+	}
+}
+
+// Ack acknowledges a message at the given position.
+func (i *Iterator) Ack(ctx context.Context, sdkPosition sdk.Position) error {
+	// if ack policy is 'none' just return nil here
+	if i.consumerInfo.Config.AckPolicy == nats.AckNonePolicy {
+		return nil
+	}
+
+	i.Lock()
+	defer i.Unlock()
+
+	if err := i.canAck(sdkPosition); err != nil {
+		return fmt.Errorf("message cannot be acknowledged: %w", err)
+	}
+
+	if err := i.unackMessages[0].Ack(); err != nil {
+		return fmt.Errorf("ack message: %w", err)
+	}
+
+	// remove acknowledged message from the slice
+	i.unackMessages = i.unackMessages[1:]
+
+	return nil
+}
+
+// Stop stops the Iterator, unsubscribes from a subject.
+func (i *Iterator) Stop() (err error) {
+	if i.subscription != nil {
+		// it will delete a consumer belonged to the subscription as well
+		if err = i.subscription.Unsubscribe(); err != nil {
+			return fmt.Errorf("unsubscribe: %w", err)
+		}
+	}
+
+	close(i.messages)
+
+	if i.conn != nil {
+		i.conn.Close()
+	}
+
+	return nil
+}
+
+// getConsumerOptions returns a []nats.SubOpt slice based on the incoming params and a sdk.Position.
+func getConsumerOptions(params IteratorParams) ([]nats.SubOpt, error) {
+	opts := make([]nats.SubOpt, 0)
+
+	position, err := parsePosition(params.SDKPosition)
+	if err != nil {
+		return nil, fmt.Errorf("parse position: %w", err)
+	}
+
+	// if the position has a non-zero OptSeq
+	// the connector will start consuming from that position
+	if position.OptSeq != 0 {
+		// add 1 to the sequence in order to skip the consumed message at this position
+		// and start consuming new messages
+		// deliverPolicy in this case will become a DeliverByStartSequencePolicy.
+		opts = append(opts, nats.StartSequence(position.OptSeq+1))
+	} else {
+		switch params.DeliverPolicy {
+		case nats.DeliverAllPolicy:
+			opts = append(opts, nats.DeliverAll())
+		case nats.DeliverNewPolicy:
+			opts = append(opts, nats.DeliverNew())
+		}
+	}
+
+	switch params.AckPolicy {
+	case nats.AckAllPolicy:
+		opts = append(opts, nats.AckAll())
+	case nats.AckExplicitPolicy:
+		opts = append(opts, nats.AckExplicit())
+	case nats.AckNonePolicy:
+		opts = append(opts, nats.AckNone())
+	}
+
+	opts = append(opts, []nats.SubOpt{
+		nats.Durable(params.Durable),
+		nats.ReplayInstant(),
+		nats.DeliverSubject(fmt.Sprintf("%s.%s", params.Durable, params.Stream)),
+		nats.EnableFlowControl(),
+		nats.IdleHeartbeat(heartbeatTimeout),
+	}...)
```

Comment:
Is there a special reason why we put the options in a slice, and then use the ellipsis operator? i.e. why not have:
```go
	opts = append(opts, 
		nats.Durable(params.Durable),
		nats.ReplayInstant(),
		nats.DeliverSubject(fmt.Sprintf("%s.%s", params.Durable, params.Stream)),
		nats.EnableFlowControl(),
		nats.IdleHeartbeat(heartbeatTimeout),
	)
```
?

Reply:
No, there's no special reason for this, I've removed the redundant slice, thank you